### PR TITLE
tsduck: init at 3.31-2761

### PR DESCRIPTION
**Pull request by siriobalmelli**
Signed-off-by: Sirio Balmelli <sirio@b-ad.ch>

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin

### CODE REVIEW

File: pkgs/tools/video/tsduck/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tsduck";
+  version = "v3.31-2761";
+
+  src = fetchFromGitHub {
+    owner = "tsduck";
+    repo = pname;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/tools/video/tsduck/default.nix:15]:
```suggestion
  pname = "tsduck";
  version = "3.31-2761";

  src = fetchFromGitHub {
    owner = "tsduck";
    repo = pname;
    rev = "v${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/video/tsduck/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  enableParallelBuilding = true;
+  makeFlags = [
+    "NODEKTEC=1"
+    "NOHIDES=1"
+    "NOPCSC=1"
+    "NORIST=1"
+    "NOVATEK=1"
+  ] ++ installFlags;
+
+  doCheck = false;  # calls out to https://tsduck.io/download/test/...
```
Comment (viraptor) [inline pkgs/tools/video/tsduck/default.nix:35]: That seems to be the case for 4 tests out of a large suite. Could you disable those tests instead?

Comment (viraptor) [reply pkgs/tools/video/tsduck/default.nix:35]: For example before testing, substitute:
```
    TSUNIT_TEST(testMasterPlaylist);
    TSUNIT_TEST(testMasterPlaylistWithAlternate);
    TSUNIT_TEST(testMediaPlaylist);
```
with empty text in https://github.com/tsduck/tsduck/blob/eddb9c7a4f7f3aeac40713758f550c98426b148d/src/utest/utestHLS.cpp#L57-L59

Comment (siriobalmelli) [reply pkgs/tools/video/tsduck/default.nix]: Very good point, thank you.

I've added directives to remove the failing tests in the patch phase.

Fixed up and respun with all changes.

**Comment by onny** [comment]
Please change PR and commit title to `tsduck: init at 3.31-2761` without the `v`-prefix

File: pkgs/tools/video/tsduck/default.nix [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    "install-tools"
+    "install-devel"
+  ];
+
+  meta = with lib; {
+    description = "The MPEG Transport Stream Toolkit";
+    homepage    = "https://github.com/tsduck/tsduck";
+    license     = licenses.bsd2;
+    maintainers = with maintainers; [ siriobalmelli ];
+    platforms   = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/video/tsduck/default.nix:81]:
```suggestion
    platforms   = platforms.all;
    # requires Xcode
    broken = stdenv.isDarwin;
```

**Comment by onny** [comment]
Result of `nixpkgs-review pr 190692` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tsduck</li>
  </ul>
</details>

**Comment by onny** [comment]
Thank you!

File: pkgs/tools/video/tsduck/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/tools/video/tsduck/default.nix:81]: ofborg can't build it and then hydra can't either https://logs.nix.ci/?key=nixos/nixpkgs.190692&attempt_id=e3b6cd39-2563-4145-a705-9d00cb2bf626

**Pull request merged**